File: core/src/org/tetawex/ecf/util/LevelDataUtils.java
package org.tetawex.ecf.util;

import com.badlogic.gdx.utils.Json;
import org.tetawex.ecf.model.LevelData;

/**
 * Created by tetawex on 16.08.17.
 */
public class LevelDataUtils {
    public static String toJson(LevelData object){
        return JsonUtils.getSerializer().toJson(new SavableLevelData(object));
    }
    public static LevelData fromJson(String json){
        return JsonUtils.getSerializer().fromJson(json, SavableLevelData.class).createLevelData();
    }
}

File: core/src/org/tetawex/ecf/util/LevelDataUtils.kt
package org.tetawex.ecf.util

import com.badlogic.gdx.utils.Json
import org.tetawex.ecf.model.LevelData

/**
 * Created by tetawex on 16.08.17.
 */
object LevelDataUtils {
    fun toJson(data: LevelData): String {
        return JsonUtils.serializer.toJson(SavableLevelData(data))
    }

    fun fromJson(json: String): LevelData {
        return JsonUtils.serializer.fromJson(json, SavableLevelData::class.java).createLevelData()
    }
}
